File: Applied Course/4.Problem Solving/4.Problems on Stacks and Queues/9.Petrol Filling Problem Simple Solution.c
#include<stdio.h>
#include<stdlib.h>

int getTour(int petrol[],int distance[], int n)
{
	int sum=0,start=0,diff=0;
        for(int i=0;i<n;i++)
        {
            sum=sum+petrol[i]-distance[i];
            
            if(sum<0)
            {
                start=i+1;//changing start point
                diff+=sum;//storing the negative values
                sum=0;//starting again form new station
            }
        }
        return sum+diff>0? start: -1;//if sum+diff <0 which means the petrol and distance is not matched
}

int main()
{
	int petrol[4]={4,6,7,4};
	int distance[4]={6,5,3,5}; 
  
  	int value=getTour(petrol,distance,4);
  	printf("%d\n",value);
}

File: Applied Course/4.Problem Solving/4.Problems on Stacks and Queues/9.Petrol Filling Problem Simple Solution.py
def getTour(petrol,distance,n):
    s,start,diff=0,0,0
    for i in range(n):
        s=s+petrol[i]-distance[i]
        if(s<0):
            start=i+1
            diff+=s
            s=0
    if s+diff>0:
        return start
    else:
        return -1
    
if __name__=='__main__':
    petrol=[4,6,7,4]
    distance=[6,5,3,5]
    value=getTour(petrol,distance,4)
    print(value)
